### PR TITLE
gcp-bastion

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/roadmap/issues/919

### CODE REVIEW

**Comment by calvix** [comment]
sorry not ready for review yet, should be

**Comment by calvix** [comment]
LFR

 looks like it's working fine,  bastion is created another machine via MachineDeployment, and with custom pre-init-command we never allow kubeadm to execute so it never joins the cluster
 
 I still need to changes for opsctl but that is another story

File: helm/cluster-gcp/templates/_bastion.tpl [hunk 1]
```diff
@@ -37,13 +25,17 @@ spec:
...
+          kind: KubeadmConfigTemplate
+          name: {{ include "resource.default.name" $ }}-bastion
       clusterName: {{ include "resource.default.name" $ }}
+      failureDomain: {{ index .Values.gcp.failureDomains 0 }}
       infrastructureRef:
         apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
         kind: GCPMachineTemplate
         name: {{ include "resource.default.name" $ }}-bastion
-      version: v0.0.0
+      version: {{ .Values.kubernetesVersion }}
```
Comment (fiunchinho) [inline helm/cluster-gcp/templates/_bastion.tpl:38]: If I understand correctly, the k8s version here doesn't really matter right?

**Review by fiunchinho — APPROVED** [review]

File: helm/cluster-gcp/templates/_bastion.tpl (hunk 1 above)
Comment (calvix) [reply helm/cluster-gcp/templates/_bastion.tpl:38]: Yeah, previously we used `0.0.0` but that is not in this case allowed, the k8s will never launch on bastion

**Pull request merged**